core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1089,11 +1089,21 @@ private void replaceSubQueries(
       RelOptUtil.Logic logic) {
     findSubQueries(bb, expr, logic, false);
     for (SubQuery node : bb.subQueryList) {
-      substituteSubQuery(bb, node);
+      substituteSubQuery(bb, node, false);
     }
   }
 
-  private void substituteSubQuery(Blackboard bb, SubQuery subQuery) {
+  private void replaceSubQueriesOnFilter(
+      final Blackboard bb,
+      final SqlNode expr,
+      RelOptUtil.Logic logic) {
+    findSubQueries(bb, expr, logic, false);
+    for (SubQuery node : bb.subQueryList) {
+      substituteSubQuery(bb, node, true);
+    }
+  }
```

Comment:
This looks like code duplication with old `replaceSubQueries`.

Can you please rename `replaceSubQueriesOnFilter` to `replaceSubQueries`, and make the old method delegate to the new one?

Reply:
ok, let me try it

Reply:
I have modified